### PR TITLE
fix: job edit no longer assigns default stage template implicitly (#1105)

### LINKED ISSUE #1105: [Jobs][Bug] Editing a job with no workflow can assign the default stage template
Status: OPEN

## What is wrong
`IOSEditJobSheet` auto-selects a stage template for jobs that currently have no `stageTemplateId`, even when the user only opens the edit sheet to change unrelated job details.

Evidence in `Weird Parts IOS/Weird Parts IOS/Features/Jobs/IOSEditJobSheet.swift`:
- `loadStageTemplates()` initializes `selectedStageTemplateId` to `job.stageTemplateId ?? default template ?? first template`.
- `saveJob()` treats any `selectedStageTemplateId != job.stageTemplateId` as a template change, shows the confirmation dialog, and on confirm calls `previewJobStageTemplateAssignment` + `assignJobStageTemplate`.

That means legacy/imported jobs or any job without an assigned workflow can be forced through the template-change path when saving a normal edit like customer name, address, notes, budget, or hours. The picker also shows the default/first workflow as selected, so the user can unintentionally assign a workflow without explicitly choosing one.

## Reproduction steps
1. Have at least one active job stage template with stages.
2. Open a job whose `stage_template_id` is NULL.
3. Open Edit Job.
4. Change an unrelated field, such as Notes or Customer Name.
5. Tap Save.

## Expected
Saving an unrelated edit should update only the changed job fields. A job with no workflow should remain workflow-less unless the user explicitly selects a workflow/template.

## Actual
The edit sheet preselects the default/first workflow and treats that as a change from NULL. Saving prompts for a stage-template change and, if confirmed, assigns the template and first/replacement stage.

## Impact / severity
Severity: Medium
Blast radius: Jobs field workflow, especially legacy/imported jobs and beta field-test job edits. This can unexpectedly alter a job's workflow/stage state while the user is only editing normal job metadata.

## Suggested fix
Track whether the workflow picker was explicitly changed by the user. For jobs with `job.stageTemplateId == nil`, either:
- keep `selectedStageTemplateId` nil and include a "No workflow" option, or
- allow showing a default suggestion but do not apply/confirm assignment unless the user changes the picker.

Add an iOS regression test that a nil-template job can save unrelated fields without setting `showingTemplateChangeConfirmation`/calling `assignJobStageTemplate`.

## Evidence from bug hunt
- Inspected recent job-edit changes in commit `222b51673` and source lines around `IOSEditJobSheet.loadStageTemplates()` / `saveJob()`.
- Searched existing open/closed GitHub issues for stage-template edit/default coverage; only the original configurable-stage request (#625) appeared, not this regression.
- Ran focused core verification: `swift test --filter JobsServiceTests` passed (139 tests). This issue is in the SwiftUI edit-sheet state/wiring layer, not the core JobsService test suite.

### PR DESCRIPTION
**Pull request by xXKillerNoobYT**
## Summary

Fixes #1105 (beta push Batch I, `docs/plans/beta-readiness-issue-resolution-2026-07.md`): opening **Edit Job** on a job with no workflow (`stage_template_id` NULL — legacy/imported jobs) auto-selected the default/first stage template, so saving an unrelated edit (notes, customer name, budget, hours...) routed through the template-change confirmation and could silently assign a workflow + stage.

Changes in `Weird Parts IOS/Weird Parts IOS/Features/Jobs/IOSEditJobSheet.swift`:

- **Baseline mirrors the saved job.** `selectedStageTemplateId` is initialized in `init` to `job.stageTemplateId` (nil included). `loadStageTemplates()` no longer falls back to the default/first template.
- **Explicit "No Workflow" state.** Workflow-less jobs show a "No Workflow" option selected in the picker (direct optional binding, no coercion to the first template). The template is only persisted when the user deliberately changes the picker — selection vs saved baseline, matching the sheet's existing changed-vs-original save idioms.
- **Deliberate assignment copy.** Assigning a workflow to a job that had none gets its own confirmation ("Assign workflow?" / starts at the first stage), distinct from the change-template dialog.
- **Removal semantics unchanged (noted per issue triage).** Jobs that already have a workflow keep a template-only picker. `JobsService.assignJobStageTemplate` takes a non-optional `templateId` — there is no unassign path in the service, and issue #1105 (no comments) does not request one, so removal stays unsupported here rather than inventing new semantics.

No core/`JobsService` changes — the bug was entirely in the SwiftUI edit-sheet state wiring.

## Testing

- New regression test `testEditingWorkflowLessJobDoesNotDraftDefaultStageTemplate` in `Weird Parts IOS/Weird Parts IOSTests/IOSEditJobSheetRegressionTests.swift`: asserts the nil-preserving baseline, the explicit "No Workflow" nil-tag option, absence of both default-fallback bug patterns, and that assignment stays behind the explicit confirmation path. All asserted strings verified against code lines (not comments).
- Also repaired a stale assertion in the same suite: it still expected `guard Double(trimmed) != nil else`, which a3acafac6 (#1346) rewrote to `guard let number = Double(trimmed), number.isFinite else` — the suite was red on main before this PR.
- `xcrun swiftc -parse` clean on both edited files.
- Core untouched; issue author's focused run of `swift test --filter JobsServiceTests` (139 tests) was already green against unmodified core.

## CI routing

PR gates run on **ubuntu-latest** (cloud); Xcode/iOS build + test verification stays on the **local Mac runner** (hybrid CI routing per repo policy).

Closes #1105

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: Weird Parts IOS/Weird Parts IOS/Features/Jobs/IOSEditJobSheet.swift [hunk 1]
```diff
@@ -128,16 +133,24 @@ struct IOSEditJobSheet: View {
...
+                            // have a workflow keep the template-only list — the
+                            // service has no unassign path, so removal stays
+                            // unsupported here.
+                            if job.stageTemplateId == nil {
+                                Text("No Workflow").tag(Int64?.none)
+                            }
                             ForEach(stageTemplates.filter { $0.stageCount > 0 }) { template in
-                                Text(template.name).tag(template.id)
+                                Text(template.name).tag(Int64?.some(template.id))
                             }
```
[the remark this answers is not in the document]
Comment (xXKillerNoobYT) [reply Weird Parts IOS/Weird Parts IOS/Features/Jobs/IOSEditJobSheet.swift]: Fixed in fefb5ce4f — the Picker iterates stageTemplates directly; the stageCount filter lives only in loadStageTemplates().

File: Weird Parts IOS/Weird Parts IOS/Features/Jobs/IOSEditJobSheet.swift [hunk 2]
```diff
@@ -128,16 +133,24 @@ struct IOSEditJobSheet: View {
                             .font(.caption)
```
[the remark this answers is not in the document]
Comment (xXKillerNoobYT) [reply Weird Parts IOS/Weird Parts IOS/Features/Jobs/IOSEditJobSheet.swift:136]: Fixed in fefb5ce4f — empty-state copy now says "No stage templates with stages available" so stage-less templates aren't misreported as absent.
